### DISCUSSION TITLE
Multiple default keymaps?

### DISCUSSION THREAD

**Discussion by manujarvinen** [helix-editor/helix#7932] [Ideas]
New user here, new in the Vim, NeoVim, Emacs world as well.

In Blender 3D software one can choose between 'Default Blender' keymap or other defaults that are for people coming from different 3D packages.

I can see that Helix tries to keep at least some keys familiar for Vim users. But **how about offering a separate invented-from-scratch-Helix-default keymap** as a choice at Helix first launch or installation for people who are totally new in these keyboard-shortcut-heavy text editors which Helix is? 

The Vim-influenced-Helix keymap could be for only those who are transitioning from Vim, right?

You say that:

> Vision: We aren't playing code golf. It's more important for the keymap to be consistent and easy to memorize than it is to save a key stroke or two when editing.

But then you have things like: 

```
In INSERT mode:
Ctrl-w   =   Delete previous word   =   delete_word_backward
Alt-d    =   Delete next word       =   delete_word_forward
Ctrl-h   =   Delete previous char   =   delete_char_backward
Ctrl-d   =   Delete next char       =   delete_char_forward
```

Why not go all the way and make it VERY easy to remember, like so:

```
In INSERT mode:
A-h     =   Backward a char           =   move_char_left
A-l     =   Forward a char            =   move_char_right
C-h     =   Move previous word start  =   move_prev_word_start
C-l     =   Move next word start      =   move_next_word_start
C-A-h   =   Delete previous char      =   delete_char_backward
C-A-l   =   Delete next char          =   delete_char_forward
C-S-h   =   Delete previous word      =   delete_word_backward
C-S-l   =   Delete next word          =   delete_word_forward
C-A-S-h =   Move to line start        =   goto_line_start
C-A-S-l =   Move to line end          =   goto_line_end_newline
```

This is just an example. 
But I hope you see what I mean.
I realize there are quite a lot of modifiers in use in the suggestion.
But I, for one, don't like if I need to whack into NORMAL mode multiple times in a row.

In my opinion, there certainly is room for more logic in the keymap.

Okay, my example is not the best one out there, but the main question remains: **how about offering a separate invented-from-scratch-Helix-default keymap and not copying Vim-shortcuts into that unless it just makes perfect sense?**

**Comment by pascalkuthe** [comment]
We are not interested in adding vim-like keymaos. It's not just "things are bound to different keys by default but  instead the editor working fundamentally differently and we are not interested in emulating vim. See #4419 and https://github.com/helix-editor/helix/issues/4419#issuecomment-1460448926 in particular
